test/deadline_submitter_for_unreal/unit/cases/test_shot_chunks.py

```diff
@@ -0,0 +1,52 @@
+import sys
+import unittest
+from unittest.mock import patch, MagicMock
+
+from deadline.unreal_submitter.unreal_open_job.job_step import RenderJobStep
+
+
+class TestShotChunks(unittest.TestCase):
+
+    @staticmethod
+    def helper_test_chunk_size(chunk_size: int, shots_count: int, expected_chunk_id: list[int]):
+        render_step = RenderJobStep(
+            step_template={
+                'parameterSpace': {
+                    'taskParameterDefinitions': [
+                        {'name': 'ChunkSize', 'type': 'INT'},
+                        {'name': 'ChunkId', 'type': 'INT'}
+                    ]
+                }
+            },
+            step_settings=MagicMock(),
+            host_requirements=MagicMock(),
+            queue_manifest_path=MagicMock(),
+            shots_count=shots_count,
+            task_chunk_size=chunk_size
+        )
+        parameters = render_step._job_step['parameterSpace']['taskParameterDefinitions']
+
+        chunk_size_param = next((p for p in parameters if p['name'] == 'ChunkSize'), None)
+        assert chunk_size_param is not None
+        assert chunk_size_param['range'][0] == chunk_size
+
+        chunk_id_param = next((p for p in parameters if p['name'] == 'ChunkId'), None)
+        assert chunk_id_param is not None
+        assert chunk_id_param['range'] == expected_chunk_id
+
+    def test_chunk_size(self):
+        # TODO find better approach on parametrizing unittest.TestCase cases
```

Comment:
IMO this is one of the reasons to prefer pytest over unittest - I know we've got a bunch of unittest.TestCases around in this repo already though.  I've been holding off on reviewing since I know you were looking at at least a few more changes from our offline discussion - is there more test coverage coming for these changes though, was that pending this TODO getting sorted out?

Reply:
We're not planning yet on increasing the test coverage of the Python side for the recent changes since other changes affect YAML templates and adaptor side that already covered with tests.